src/lib/sql/sql_to_ast_translator.cpp

```diff
@@ -844,6 +860,48 @@ std::shared_ptr<AbstractASTNode> SQLToASTTranslator::_translate_show(const hsql:
   return {};
 }
 
+std::shared_ptr<AbstractASTNode> SQLToASTTranslator::_translate_create(const hsql::CreateStatement& create_statement) {
+  switch (create_statement.type) {
+    case hsql::CreateType::kCreateView: {
+      auto view = _translate_select((const hsql::SelectStatement&)*create_statement.select);
+      if (create_statement.viewColumns) {
+        std::vector<std::shared_ptr<Expression>> projections;
+        ColumnID column_id{0};
+        Assert(create_statement.viewColumns->size() == view->output_col_count(),
+               "Number of Columns in CREATE VIEW does not match SELECT statement");
+        for (const auto& alias : *create_statement.viewColumns) {
+          // rename columns so they match the view definition
+          projections.push_back(Expression::create_column(column_id, alias));
+          ++column_id;
+        }
+        auto projection_node = std::make_shared<ProjectionNode>(projections);
+        projection_node->set_left_child(view);
+        view = projection_node;
+      }
+
+      StorageManager::get().add_view(create_statement.tableName, view);
+      return std::make_shared<EmptyNode>();
+    }
+    default:
+      Fail("hsql::CreateType is not supported.");
+  }
+
+  return {};
+}
+
+std::shared_ptr<AbstractASTNode> SQLToASTTranslator::_translate_drop(const hsql::DropStatement& drop_statement) {
+  switch (drop_statement.type) {
+    case hsql::DropType::kDropView: {
+      StorageManager::get().drop_view(drop_statement.name);
+      return std::make_shared<EmptyNode>();
```

Comment:
Because we are generating `EmptyNode`s here, we cannot really have any follow-up logic, e.g., having the query fail because that view does not exist.

Reply:
The logic happens in the line above. If the view does not exist, the SM throws an exception.

Reply:
My idea is that there are two types of SQL statements: Those that end up in a physical, optimizable execution plan (e.g., SELECT, INSERT) and those that are of DDL nature (CREATE, ALTER, ...). It makes sense to have the SQLToASTTranslator deal with both, because for some, like the join, we still need the AST.

Are you questioning this concept in general (please feel free to do so) or are you questioning the interface of the EmptyNode?

Reply:
I am questioning the idea of `EmptyNode`, or rather the idea of DDL statements translating to `EmptyNode`, because I think that even DDL statements should be executed in operators